sdlib/d/gc/collector.d

```diff
@@ -195,7 +195,14 @@ private:
 			auto delta = nextTarget - lastHeapSize;
 			delta -= delta >> lgTargetDecay;
 			delta += lastHeapSize >> (lgTargetDecay + lgMinOverhead);
-			nextTarget = lastHeapSize + delta;
+
+			auto newTarget = lastHeapSize + delta;
+			if (newTarget == nextTarget) {
+				// Limit reached.
```

Comment:
You need to update the `lastTargetAdjustement` properly here, or we'll maximally adjust for decade before reaching steady state.

Reply:
Most likely it’s going to set the lastTargetAdjustment after the scan

but even if it doesn’t, the state won’t change the next time. It will do at most 1 of these loops.